astropy/time/core.py

```diff
@@ -518,8 +519,9 @@ def strptime(cls, time_string, format_string, **kwargs):
                              op_dtypes=[time_array.dtype, 'U30'])
 
         for time, formatted in iterator:
-            time_tuple = strptime(to_string(time), format_string)
-            formatted[...] = '{:04}-{}-{}T{}:{}:{}'.format(*time_tuple)
+            strped_time = _strptime._strptime(to_string(time), format_string)
+            time_tuple = strped_time[0][:6] + (strped_time[1],)
+            formatted[...] = '{:04}-{}-{}T{}:{}:{}.{}'.format(*time_tuple)
```

Comment:
This needs to have a format specifier for the microseconds spot of `{:06}`, otherwise of a value of 1 for microsecs will result in a time like `12:23:34.1` instead of `12:23:34.000001`.  You should adjust your tests accordingly, which currently only pass accidentally.

For that matter, better to make all the format specifiers use `{:02}`.  The Time parser is lenient and accepts non-compliant ISOT values like 2018-1-1, but we shouldn't output dates like that even internally.